Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php

```diff
@@ -21,236 +23,151 @@
+                    /* , 'phone' => $data['billing']['phone'] */
+                    , 'email' => $data['billing']['email']
+                    , 'tags' => []
+                    ];
+
+        /* Extract the items details. */
+        $items = array();
+        foreach ( $order->get_items() as $item ) {
+          $items[] = [ 'item' => $item['name']
+                    , 'units' =>  $item['quantity']
+                    , 'unitPrice' => number_format( number_format( ( float )$item['subtotal'], 2) / number_format( ( float )$item['quantity'], 2 ), 2 )
+                    /* , 'type' => '' */
+                    /* , 'code' => '' */
+                    /* , 'vatPercent' => '' */
+                    /* , 'itemDescription' => '' */
+                    ];
         }
-        
-        // Checksum
-        ksort($inputs);
-        $query = http_build_query($inputs);
-        $encoded = hash_hmac('sha512', $query, $privateKEY, true);
-        $checksum = '';
-        $checksum = base64_encode($encoded);
-        
+
+        /* Build the data object to be posted to Twispay. */
+        $orderData = [ 'siteId' => $siteID
+                    , 'customer' => $customer
+                    , 'order' => [ 'orderId' => (isset( $_GET['tw_reload'] ) && $_GET['tw_reload']) ? ($data['id'] . '_' . time()) : ($data['id'])
+                                  , 'type' => 'purchase'
+                                  , 'amount' => $data['total']
+                                  , 'currency' => $data['currency']
+                                  , 'items' => $items
+                                  , 'tags' => []
+                                  /* , 'intervalType' => '' */
+                                  /* , 'intervalValue' => 1 */
+                                  /* , 'trialAmount' => 1 */
+                                  /* , 'firstBillDate' => '' */
+                                  /* , 'level3Type' => '', */
+                                  /* , 'level3Airline' => [ 'ticketNumber' => '' */
+                                  /*                      , 'passengerName' => '' */
+                                  /*                      , 'flightNumber' => '' */
+                                  /*                      , 'departureDate' => '' */
+                                  /*                      , 'departureAirportCode' => '' */
+                                  /*                      , 'arrivalAirportCode' => '' */
+                                  /*                      , 'carrierCode' => '' */
+                                  /*                      , 'travelAgencyCode' => '' */
+                                  /*                      , 'travelAgencyName' => ''] */
+                                  ]
+                    , 'cardTransactionMode' => 'authAndCapture'
+                    , 'cardId' => 0

```

Comment:
We should comment this line as `cardId` is not a required field (and setting it to `0` value means it will not be used).

Reply:
Done